src/frontend/src/eth/components/send/SendAmount.svelte

```diff
@@ -1,91 +1,92 @@
 	import type { Token } from '$lib/types/token';
+	import SendInputAmount from '$lib/components/send/SendInputAmount.svelte';
+	import { InsufficientFundsError } from '$lib/types/send';
+	import { getMaxTransactionAmount } from '$lib/utils/token.utils';
 
 	export let amount: number | undefined = undefined;
 	export let insufficientFunds: boolean;
 	export let nativeEthereumToken: Token;
 
-	let insufficientFundsError: string | undefined;
+	let insufficientFundsError: InsufficientFundsError | undefined = undefined;
 
 	$: insufficientFunds = nonNullish(insufficientFundsError);
 
 	const { feeStore: storeFeeData } = getContext<FeeContext>(FEE_CONTEXT_KEY);
 	const { sendTokenDecimals, sendBalance, sendTokenId } = getContext<SendContext>(SEND_CONTEXT_KEY);
 
-	const validate = () => {
-		if (invalidAmount(amount)) {
-			insufficientFundsError = undefined;
+	let fee: BigNumber | undefined | null = undefined;
+
+	let timer: NodeJS.Timeout | undefined;
+
+	$: $storeFeeData,
+		(() => {
+			timer = setTimeout(() => {
+				fee = getFee($storeFeeData);
+			}, 500);
+		})();
+
+	onDestroy(() => {
+		if (isNullish(timer)) {
 			return;
 		}
 
+		clearInterval(timer);
+	});
+
+	$: customValidate = (userAmount: BigNumber) => {
 		if (isNullish($storeFeeData)) {
-			insufficientFundsError = undefined;
 			return;
 		}
 
-		const userAmount = parseToken({
-			value: `${amount}`,
-			unitName: $sendTokenDecimals
-		});
-
 		// If ETH, the balance should cover the user entered amount plus the min gas fee
 		if (isSupportedEthTokenId($sendTokenId)) {
 			const total = userAmount.add(minGasFee($storeFeeData));
 
 			if (total.gt($sendBalance ?? BigNumber.from(0n))) {
-				insufficientFundsError = $i18n.send.assertion.insufficient_funds_for_gas;
-				return;
+				return new InsufficientFundsError($i18n.send.assertion.insufficient_funds_for_gas);
 			}
 
-			insufficientFundsError = undefined;
 			return;
 		}
 
 		// If ERC20, the balance of the token - e.g. 20 DAI - should cover the amount entered by the user
 		if (userAmount.gt($sendBalance ?? BigNumber.from(0n))) {
-			insufficientFundsError = $i18n.send.assertion.insufficient_funds_for_amount;
-			return;
+			return new InsufficientFundsError($i18n.send.assertion.insufficient_funds_for_amount);
 		}
+
 		// Finally, if ERC20, the ETH balance should be less or greater than the max gas fee
 		const maxFee = maxGasFee($storeFeeData);
 		const ethBalance = $balancesStore?.[nativeEthereumToken.id]?.data ?? BigNumber.from(0n);
-
 		if (nonNullish(maxFee) && ethBalance.lt(maxFee)) {
-			insufficientFundsError = $i18n.send.assertion.insufficient_ethereum_funds_to_cover_the_fees;
-			return;
+			return new InsufficientFundsError(
+				$i18n.send.assertion.insufficient_ethereum_funds_to_cover_the_fees
+			);
 		}
-
-		insufficientFundsError = undefined;
 	};
 
-	const debounceValidate = debounce(validate);
-
-	$: amount, $storeFeeData, debounceValidate();
+	let calculateMax = (): number | undefined => {
```

Comment:
`getMaxTransactionAmount({` returns `number`, is `number | undefined` correct?

Reply:
corrected